packages/muelu/test/scaling/MatrixMatrixMultiply.cpp

```diff
@@ -209,6 +209,10 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib &lib, int ar
     options.print_warnings = false;
     timer->report(std::cout, comm, options);
 
+    auto xmlOut = timer->reportWatchrXML(std::string("MueLu_MatrixMatrix") + std::to_string(comm->getSize()), comm);
```

Comment:
Why would the XML string have zero length?  Is that some sort of indicator that you didn't output Watchr XML?

Reply:
Also, this interface looks like a function, but I'm guessing that it has side effects (e.g., output to a file).  Why not just have it return `void` in that case?

EDIT: I can tell from below that it returns the filename.  You could make that clear in the test:
```cpp
if (comm->getRank() == 0) {
  std::cout << "Printed Watchr performance report to file \"" << xmlOut << "\"" << std::endl;
}
```

Reply:
@mhoemmen Could you clarify what you mean by the test? The idea was, user code only needs to test ``xmlOut.length() != 0`` because only ranks that actually wrote a file return a non-empty string. ``if(comm->rank() == 0 && xmlOut.length() != 0)`` is certainly correct, and even though the rank test is redundant, it might make things clearer. Would you prefer I add that?

Reply:
Here's the doxygen for that function:
```
  /**
   * Dump all the data from all the MPI ranks to <code>$WATCHR_PERF_DIR/name_$DATESTAMP.xml</code>,
   *  if the environment variable WATCHR_PERF_DIR is defined and non-empty (otherwise, do nothing).
   *  DATESTAMP is calculated from the current UTC time, in the format YYYY_MM_DD.
   * @param [in] name - Name of performance test
   * @param [in] comm - Teuchos comm pointer
   * @return If on rank 0 and output was produced, the complete output filename. Otherwise the empty string.
   */
  std::string reportWatchrXML(const std::string& name, Teuchos::RCP<const Teuchos::Comm<int> > comm);
```